### PR TITLE
Adding migration section

### PR DESCRIPTION
**Pull request by guillaumebriday**
## What is the new behavior?

Trying to help people to migrate from Netlflix/fast_jsonapi.

Did I forget something ?

### CODE REVIEW

**Comment by stas** [comment]
Thanks @guillaumebriday, this is great! :bowing_man: 

I don't recall anything else (maybe the cache support changes https://github.com/jsonapi-serializer/jsonapi-serializer/pull/52/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

But I'm happy to merge this as the first step. Let me know if it's ready @guillaumebriday

**Comment by guillaumebriday** [comment]
Nice catch, will add it.

Any other suggestion? 🤔

**Comment by tonydehnke** [comment]
My 'base_serializer.rb' file has a line:

`require 'fast_jsonapi'`

Might be obvious to most, but maybe mention changing it to:
`require 'jsonapi-serializer'`

**Comment by guillaumebriday** [comment]
thanks @tonydehnke I don't need `require` in my applications, I will update the PR !

**Comment by guillaumebriday** [comment]
I think it's ready now 👍

File: README.md [hunk 1]
```diff
@@ -706,6 +707,49 @@ This gem provides the next features alongside deserialization:
...
++ require 'jsonapi-serializer'
+```
+
+### Update your cache options
+
+See [docs](https://github.com/jsonapi-serializer/jsonapi-serializer#caching).
+
+```diff
+- cache_options enabled: true, cache_length: 12.hours
++ cache_options store: Rails.cache, namespace: 'fast-jsonapi', expires_in: 1.hour
```
Comment (tonydehnke) [inline README.md:750]: Does it make sense to leave the namespace as 'fast-jsonapi' or should it get changed to something more like the gem name ie 'jsonapi-serializer'?

Comment (guillaumebriday) [reply README.md:750]: You're right! Could you open a PR?

Comment (tonydehnke) [reply README.md:750]: I can try :)

Comment (tonydehnke) [reply README.md:750]: Done in #143

**Comment by jrochette** [comment]
Hi all, I'm not sure if this is the right place to ask, so let me know and I'll be happy to move this question. I tried migrating from `fast_jsonapi` today and I hit an issue with relations.
I have this serializer (simplified for the purpose of this example):
```
class OrganizationSerializer < ApplicationSerializer
  has_many :engagements
  has_many :invites
  [...]
end
```
Which used to work, but stopped working after migration to `jsonapi-serializer 2.1.0`. I am now getting this error
```
NameError:
  OrganizationSerializer cannot resolve a serializer class for 'invite'.  Attempted to find 'InviteSerializer'. Consider specifying the serializer directly through options[:serializer].
```
As the error indicates, I don't have an `InviteSerializer` class. I guess this is something that is now required? If so, it should probably be mentioned in the migration doc?

**Comment by stas** [comment]
Hey @jrochette 

We had previous reports related to this change, one of them here:
https://github.com/jsonapi-serializer/jsonapi-serializer/issues/151#issuecomment-733591756

Would you be kind to take a look at it and let me know if that makes any sense?! In a nutshell, you might need to specify the serializer inline if it doesn't follow a conventional naming pattern.

**Comment by jrochette** [comment]
Hey @stas

Thanks for the quick response. I looked at #151 and I don't think it is the same issue, but the explanation does make sense. I agree that it's a good idea to make the serializer resolving rely less on _magic_. 

Just to put more context, I my case, it is not that the serializer is not resolved because of a namespace issue. It doesn't get resolved because it does not exist. I assume there was a mechanism in `fast_jsonapi` to do something when no serializer was found (some kind of generic serializer maybe?).

We have put our migration to `jsonapi-serializer` on hold for now, as creating and testing new serializers for models which previsouly had no defined serializer is more work than anticipated. We'll get around to create those serializers eventually. Thanks!

**Comment by stas** [comment]
> Just to put more context, I my case, it is not that the serializer is not resolved because of a namespace issue. It doesn't get resolved because it does not exist. I assume there was a mechanism in `fast_jsonapi` to do something when no serializer was found (some kind of generic serializer maybe?).

That's actually excluded. What could have been the case is that you'd get another/wrong serializer class doing the serialization for you.

Good luck!

**Pull request merged**